pkg/asset/manifests/gcp/cluster.go

```diff
@@ -33,26 +38,44 @@ func GenerateClusterAssets(installConfig *installconfig.InstallConfig, clusterID
 		networkName = installConfig.Config.GCP.Network
 	}
 
-	masterSubnet := gcp.DefaultSubnetName(clusterID.InfraID, "master")
+	masterSubnetName := gcp.DefaultSubnetName(clusterID.InfraID, "master")
+	masterSubnetCidr := ""
 	if installConfig.Config.GCP.ControlPlaneSubnet != "" {
-		masterSubnet = installConfig.Config.GCP.ControlPlaneSubnet
+		masterSubnetName = installConfig.Config.GCP.ControlPlaneSubnet
+
+		masterSubnet, err := getSubnet(installConfig.Config.GCP.NetworkProjectID, installConfig.Config.GCP.Region, masterSubnetName)
+		if err != nil || masterSubnet == nil {
```

Comment:
Instead of checking for `masterSubnet == nil` here, why not have `getSubnet` return an error in that case?